examples/build_docker.sh

```diff
@@ -3,25 +3,42 @@
 # Exit script as soon as a command fails.
 set -ex
 
+# Predefined ANSI escape codes for colors
+YELLOW='\033[0;33m'
+NC='\033[0m'
+
 NAME="$1"
 
 # Switch to current directory (./examples) then out to root for specific examples
 pushd $(dirname ${BASH_SOURCE[0]})
 cd ../
 
+# Pick the correct tag to pull from Docker Hub based on OS architecture
+_warning="
+${YELLOW}WARNING${NC}: You are building smart contracts using ARM64. The resulting artifacts will
+be usable for testing, but won't pass the CI check for inclusion in master due to the
+reproducibility requirements.
+"
+if [[ $(uname -m) == 'arm64' ]]; then
+    echo -e "$_warning"
```

Comment:
Do you know a sane way to declare a multiline string literal inside the if without indentation getting in the way? That's the only reason I have put it outside.

Reply:
Not sure! I see some options in this https://stackoverflow.com/questions/23929235/multi-line-string-with-extra-space-preserved-indentation but none seem that clean.

I wouldn't worry about it too much, just pointing out this could be confusing for a developer without context